### PR TITLE
[BugFix] Support to access object storage in https

### LINKED ISSUE #7204: Access S3 object with error message "IO error: No response body"
### Steps to reproduce the behavior (Required)

1. Create hive table which stores files on s3.
2. Create external hive table and query it

And s3 bucket has some configurations like following: which means it can be accessed only in ssl way.
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Statement1",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::starrocks-tpch100g",
                "arn:aws:s3:::starrocks-tpch100g/*"
            ],
            "Condition": {
                "Bool": {
                    "aws:SecureTransport": "false"
                }
            }
        }
    ]
}
```

### Expected behavior (Required)
Return results

### Real behavior (Required)
Return error message "IO error: No response body"

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7204 

## Problem Summary(Required) ：

When user running SR on ec2 and ec2 has proper IAM role, there is no need to specify 
- AK/SK
- endpoint

But when in that case, we are not sure if bucket/object can be accessed in http(perhaps there are policies there).  So we add a option `object_storage_endpoint_use_ssl`, to let user specify that.

So basically there are two ways of configuration. When running on non-ec2 instance

```
object_storage_access_key_id = xxx
object_storage_secret_access_key = xxx
object_storage_endpoint = https://s3.us-east-1.amazonaws.com
object_storage_max_connection = 102400
AWS_EC2_METADATA_DISABLED=true
```

When runing on ec2 instance

```
object_storage_max_connection = 102400
object_storage_endpoint_use_ssl = true/false
```

### CODE REVIEW

**Review by DorianZheng — COMMENTED** [review]
@sduzh PTAL

**Review by DorianZheng — COMMENTED** [review]
LGTM

File: be/src/fs/fs_s3.cpp [hunk 1]
```diff
@@ -130,11 +130,16 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const ClientConfigurati
...
         config.endpointOverride = uri.endpoint();
     } else if (!config::object_storage_endpoint.empty()) {
         config.endpointOverride = config::object_storage_endpoint;
+    } else {
+        // To use implicit endpoint.
+        config.scheme = Aws::Http::Scheme::HTTP;
+        if (config::object_storage_endpoint_use_https) {
+            config.scheme = Aws::Http::Scheme::HTTPS;
+        }
     }
```
Comment (sduzh) [inline be/src/fs/fs_s3.cpp:143]:
```suggestion
    } else if (config::object_storage_endpoint_use_https) {
         config.scheme = Aws::Http::Scheme::HTTPS;
    } else {
        config.scheme = Aws::Http::Scheme::HTTP;
    }
```

**Review by sduzh — APPROVED** [review]

**Comment by dirtysalt** [comment]
run starrocks_be_unittest

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Comment by dirtysalt** [comment]
run starrocks_clang-format

**Review by caneGuy — APPROVED** [review]
